scripts/generate-sdf.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs')
+const path = require('path')
+const mkdirp = require('mkdirp')
+const fontnik = require('fontnik')
+
+const font = fs.readFileSync(
+  path.join(__dirname, '../fonts/opensans/OpenSans-Regular.ttf')
+)
+
+const outDir = path.join(__dirname, '../sdf/opensans')
+mkdirp.sync(outDir)
+
+generateSdf({ font, outDir })
+
+async function generateSdf({ font, outDir }) {
+  return new Promise((resolve, reject) => {
+    for (let i = 0; i < 65536; i = i + 256) {
```

Comment:
for my own education, what are these numbers?

Reply:
These number are character codes. The [mapbox api](https://docs.mapbox.com/api/maps/fonts/) expects SDFs to be served as combined files of 256 glyphs (see the API details). For ASCII the map only needs to request the 0-255.pbf, but when the client library (mapbox-gl-js) comes across labels with other characters, it downloads the glyph ranges needed to render those characters.